### PR TITLE
Feature: [MyCollection] Add global loading / error handing

### PR DESCRIPTION
**Pull request by damassi**
The type of this PR is: **Feature**

### Description

This PR roughs in some patterns around global loading and error handling while performing different mutation operations on MyCollection. 

The flow:

- The navigation model listens for various artwork actions (add / edit / delete) and toggles `isLoading` to true
- Views can then tap into the `isLoading` state and display a loading indicator, etc

Note that I tried to take advantage of the `LoadingModal` component in Eigen but ran into a hard blocker -- since most of our operations take place modally, from within the FancyModal component, we can't display the `LoadingModal` because of a hard iOS limitation around displaying only **one** modal at a time.  (cc @pvinis and @MounirDhahri)

In place of this limitation I'm just displaying "loading" text in the form for now. We can style it later. 

See [this slack thread](https://artsy.slack.com/archives/C02BAQ5K7/p1602109961092000) for more info. 

> ⚠️   Also note I'm going to start tagging CX devs on these MyCollection PRs to help spead knowledge while I transition off of the team and hand over code. 
 
loading-alert

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

### CODE REVIEW

File: src/lib/Scenes/MyCollection/State/MyCollectionNavigationModel.tsx [hunk 1]
```diff
@@ -114,16 +132,35 @@ export const MyCollectionNavigationModel: MyCollectionNavigationModel = {
...
+
+  onAddArtworkError: thunkOn(
+    (_, storeActions) => storeActions.myCollection.artwork.addArtworkComplete,
+    (actions) => {
+      actions.setIsLoading(false)
+
+      // TODO: Log error to sentry
+      Alert.alert("Error adding artwork", "TODO add better message")
+    }
+  ),
```
Comment (damassi) [inline src/lib/Scenes/MyCollection/State/MyCollectionNavigationModel.tsx:157]: Being able to listen to actions from across the store is one of the coolest / most useful features in easy peasy

**Comment by damassi** [comment]
@jonallured I'm not sure if its something up with staging metaphysics or gravity, but deleting artworks often times out (as can be seen in the screencap). I think it might have to do with staging MP as i've never seen this before, but heads up.

File: src/lib/Scenes/MyCollection/State/MyCollectionArtworkModel.tsx [hunk 2]
```diff
@@ -324,9 +393,13 @@ export const MyCollectionArtworkModel: MyCollectionArtworkModel = {
               artworkId: sessionState.artworkId,
               // Cooerce type for MP
               costMinor: Number(costMinor),
+
+              // TODO: Wire up edit in MP
+              // externalImageUrls,
```
Comment (damassi) [inline src/lib/Scenes/MyCollection/State/MyCollectionArtworkModel.tsx:398]: @jonallured - can you give MP a look tomorrow? The code added for create mutation needs to be copied over to edit mutation as well.

**Review by MounirDhahri — APPROVED** [review]

File: src/lib/Scenes/MyCollection/State/MyCollectionArtworkModel.tsx (hunk 2 above)
Comment (jonallured) [reply src/lib/Scenes/MyCollection/State/MyCollectionArtworkModel.tsx:398]: More chatter here: https://artsy.slack.com/archives/C01B2P6LJUU/p1602173381238000

**Pull request merged**